ansible_image_validation/validation-playbooks/repo-validation-check.yaml

```diff
@@ -0,0 +1,128 @@
+  lineinfile:
+    path: "{{err_folder}}/err_msgs.log"
+    line: "\n  The RHEL Image contains more than 1 client package"
+    create: yes
+    state: present 
+  when: ( rhui_package_count | int ) > 1
+
+- name: Check if offertype is equal to byol and client package is present
+  lineinfile:
+    path: "{{err_folder}}/err_msgs.log"
+    line: "\n  Client Package is present in BYOL Offertype Image. Details of the client package {{ rhui_package }}"
+    create: yes
+    state: present 
+  when: offer_type =='byol' and rhui_package_count != "0"
+
+- name: Check if the correct client package is present according to the architecture type
+  ansible.builtin.shell:
+    cmd: |
+      #!/bin/bash
+          if [[ "{{ architecture }}" -eq "arm64" ]]; then                                                                  ## This if block refers to the check for the client packages of arm64 architecture type.  The client packages of arm64 architecture is arm64  will contain arm64 as a substring.Example client package: "rhui-azure-rhel9-eus-arm64-2.3-655.noarch","rhui-azure-rhel9-arm64-2.3-655.noarch"
+                check_correctness=$( rpm -qa | grep arm64 ) 
+                if [[ -z "$check_correctness" ]]; then 
+                      echo "{{ IncorrectPackageAccToArch }}"
+                else
+                      echo "{{ CorrectPackageAccToArch }} "                
+                fi 
+          else
+                check_correctness=$( rpm -qa | grep arm64 ) 
+                if [[ -z "$check_correctness" ]]; then 
+                      echo "{{ CorrectPackageAccToArch }}"
+                else
+                      echo " {{ IncorrectPackageAccToArch }}"                                                               ## This if block refers to the check for the client packages of x86-64 architecture type. The client packages of x86-64 architecture type don't contain arm64 or x86-64 as a substring. Example client package: "rhui-azure-rhel9-eus-2.3-655.noarch","rhui-azure-rhel9-2.3-655.noarch"
+                fi 
+          fi                            
+  register: output_package_acc_architecture
+  when: offer_type !='byol'                       
+
+- debug:
+    var: output_package_acc_architecture
+
+
+- name: "Write to error msg if incorrect client package is present according to the architecture"
+  lineinfile:
+    path: "{{err_folder}}/err_msgs.log"
+    line: "\n {{ IncorrectPackageAccToArch }}"
+    create: yes
+    state: present 
+  when: ( offer_type!='byol') and ("{{ IncorrectPackageAccToArch }}" in output_package_acc_architecture.stdout_lines)  
+
+
+
+- name: Check if the client package is present according to the offertype when offertype is not equal to byol.
+  ansible.builtin.shell:
+    cmd: |
+      #!/bin/bash
+          client_package=$( rpm -qa | grep rhui )                                                                           ## Value of client package is getting calculated
```

Comment:
There appears to be non-uniformity in naming. 
Eg: client_package
clientpkgwithoutarchitecture and so on

Please use camelCase or snake_case